stable_baselines3/dqn/dqn.py

```diff
@@ -0,0 +1,213 @@
+from stable_baselines3.dqn.policies import DQNPolicy
+
+
+class DQN(OffPolicyRLModel):
+    """
+    Deep Q-Network (DQN)
+
+    Paper: https://arxiv.org/abs/1312.5602, https://www.nature.com/articles/nature14236
+    Default parameters are taken from the nature paper
+
+    :param policy: (PPOPolicy or str) The policy model to use (MlpPolicy, CnnPolicy, ...)
+    :param env: (Gym environment or str) The environment to learn from (if registered in Gym, can be str)
+    :param learning_rate: (float or callable) The learning rate, it can be a function
+        of the current progress (from 1 to 0)
+    :param buffer_size: (int) size of the replay buffer
+    :param learning_starts: (int) how many steps of the model to collect transitions for before learning starts
+    :param batch_size: (int) Minibatch size for each gradient update
+    :param gamma: (float) Discount factor
+    :param train_freq: (int) Update the model every ``train_freq`` steps.
+    :param gradient_steps: (int) How many gradient update after each step
+    :param tau: (float) the soft update coefficient ("polyak update", between 0 and 1)
+    :param target_update_interval: (int) update the target network every ``target_update_interval`` steps.
+    :param exploration_fraction: (float) fraction of entire training period over which the exploration rate is reduced
+    :param exploration_initial_eps: (float) initial value of random action probability
+    :param exploration_final_eps: (float) final value of random action probability
+    :param max_grad_norm: (float) The maximum value for the gradient clipping
+    :param tensorboard_log: (str) the log location for tensorboard (if None, no logging)
+    :param create_eval_env: (bool) Whether to create a second environment that will be
+        used for evaluating the agent periodically. (Only available when passing string for the environment)
+    :param policy_kwargs: (dict) additional arguments to be passed to the policy on creation
+    :param verbose: (int) the verbosity level: 0 no output, 1 info, 2 debug
+    :param seed: (int) Seed for the pseudo random generators
+    :param device: (str or th.device) Device (cpu, cuda, ...) on which the code should be run.
+        Setting it to auto, the code will be run on the GPU if possible.
+    :param _init_setup_model: (bool) Whether or not to build the network at the creation of the instance
+    """
+
+    def __init__(self, policy: Union[str, Type[DQNPolicy]],
+                 env: Union[GymEnv, str],
+                 learning_rate: Union[float, Callable] = 25e-5,
+                 buffer_size: int = 1000000,
+                 learning_starts: int = 50000,
+                 batch_size: Optional[int] = 32,
+                 gamma: float = 0.99,
+                 train_freq: int = 4,
+                 gradient_steps: int = 1,
+                 tau: float = 1.0,
+                 target_update_interval: int = 10000,
+                 exploration_fraction: float = 1,
```

Comment:
```suggestion
                 exploration_fraction: float = 0.1,
```

Reply:
this is because I used the hyperparameters from the nature paper https://www.nature.com/articles/nature14236/tables/1 see final exploration frame

Reply:
okay, they had 50.000.000 frames total, so I guess a fraction of 0.02 would be appropriate?

Reply:
usually, Atari is on 10M steps (40M frames because of frame-skip)
this is a fraction of the total training steps, so should be lower than 1 anyway.
I guess we will have to tune the hyperparameters ourself to make it work in 10M steps...

Unfortunately, rainbow (that compares extensions to DQN) does not reference all of them...
https://arxiv.org/abs/1710.02298